src/Engine/Spells/CastSpellInfo.cpp

```diff
@@ -992,20 +994,19 @@ void CastSpellInfoHelpers::castSpell() {
                         spellFailed(pCastSpell, LSTR_CANT_INFERNO_OUTDOORS);
                         continue;
                     }
-                    int mon_num = render->GetActorsInViewport(4096);
                     initSpellSprite(&pSpellSprite, spell_level, spell_mastery, pCastSpell);
                     pSpellSprite.uSectorID = 0;
                     pSpellSprite.field_60_distance_related_prolly_lod = 0;
                     pSpellSprite.uFacing = 0;
-                    for (uint i = 0; i < mon_num; i++) {
+                    std::vector<Actor*> actorsInViewport = render->getActorsInViewport(4096);
+                    for (Actor* actor : actorsInViewport) {
```

Comment:
You can write this just as `for (Actor* actor : render->getActorsInViewport(4096))` and it will work as you'd expect.